core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketReader.java

```diff
@@ -66,27 +97,72 @@ public long pos() {
 
   @Override
   public void close() {
-    mReader.close();
+    if (mClosed) {
+      return;
+    }
+    mClosed = true;
```

Comment:
I think it is safer unless we expect the client to call 'close' multiple times (or retry close).